chef/cookbooks/nova_dashboard/recipes/server.rb

```diff
@@ -17,6 +17,30 @@
 include_recipe "apache2::mod_wsgi"
 include_recipe "apache2::mod_rewrite"
 
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+# Multidomain can only be enabled if the API version of Keystone is
+# equal or bigger that v3.
+keystone_api_version = keystone_settings['api_version']
+keystone_internal_auth_url = keystone_settings['internal_auth_url']
+
+# TODO(aplanas): Remove this line when keystone API 'v3' works for
+# every OpenStack configuration file.
+if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+  keystone_api_version = '3'
```

Comment:
For cloud 5 and 6 yes.  This is a way to make sure that if keystone_api_version < 3 (because the user want to set v2.0 globally), the user still can use multidomain, configuring the keystone's endpoint for horizon to v3.

So basically the user can have a global v2.0 but horizon is using v3 if needed. This is a way to transition from v2 to v3. When the transition is done, we can remove this (as expressed in the TODO) and uncomment the next lines.